app/jobs/arcgis_token_job.rb

```diff
@@ -0,0 +1,12 @@
+class ArcgisTokenJob < ApplicationJob
+  queue_as :default
+
+  attr_accessor :token_keeper
+  def initialize(token_keeper = nil)
+    @token_keeper = token_keeper || ArcgisApi::TokenKeeper.new(nil, nil, nil)
```

Comment:
can we use default arguments here?

```suggestion

    def initialize(token_keeper = ArcgisApi::TokenKeeper.new)
    @token_keeper = token_keeper
```

Reply:
removed and not using constructor based on Tim's comment